src/test/scala/io/citrine/lolo/bags/MultiTaskBaggerTest.scala

```diff
@@ -45,45 +45,88 @@ class MultiTaskBaggerTest {
-          sigmaObs.foreach { s =>
-            assert(rtolLower * s > noiseLevel, "Observational StdDev getUncertainty() is too small.")
-            assert(s < rtolUpper * noiseLevel, "Observational StdDev getUncertainty() is too large.")
+
+    Seq(MultiTaskTreeLearner()).foreach { baseLearner =>
+      // These are in Seqs as a convenience for repurposing this test as a diagnostic tool.
+      Seq(128).foreach { nRows =>
+        Seq(16).foreach { nCols =>
+          Seq(2).map { n => n * nRows }.foreach { nBags =>
+            // Used for error output.
+            val configDescription =s"learner=${baseLearner.getClass().toString()}\tnRows=$nRows\tnCols=$nCols\tnumBags=$nBags"
+
+            // Count times getStdDevObs is > and < getStdDevMean
+            var countStdObsGtStdMean = 0.0
+            var countStdObsLtStdMean = 0.0
+
+            (1 to 10).foreach { _ =>
+              val trainingDataTmp = TestUtils.generateTrainingData(nRows, nCols, noise = 0.0, function = _ => 0.0, seed = rng.nextLong())
+              val trainingData = trainingDataTmp.map { x => (x._1, x._2 + noiseLevel * rng.nextDouble()) }
+              val inputs = trainingData.map(_._1)
+              val labels = trainingData.map(_._2)
+              val baggedLearner = MultiTaskBagger(baseLearner, numBags = nBags, uncertaintyCalibration = true)
+              val RFMeta = baggedLearner.train(inputs, Seq(labels)).head
+              val RF = RFMeta.getModel()
+              val results = RF.transform(trainingData.take(4).map(_._1))
+
+              val sigmaMean: Seq[Double] = results.getUncertainty(observational = false).get.asInstanceOf[Seq[Double]]
+              sigmaMean.zip(results.asInstanceOf[BaggedMultiResult].getStdDevMean().get).foreach{ case (a,b) =>
+                assert(a == b, s"Expected getUncertainty(observational=false)=getStdDevMean() for $configDescription")
+              }
+
+              val sigmaObs: Seq[Double] = results.getUncertainty().get.asInstanceOf[Seq[Double]]
+              sigmaObs.zip(results.asInstanceOf[BaggedMultiResult].getStdDevObs().get).foreach { case (a, b) =>
+                assert(a == b, s"Expected getUncertainty()=getStdDevObs() for $configDescription")
+              }
+
+              sigmaObs.zip(sigmaMean).foreach { case (sObs, sMean) =>
+                // Uncomment for diagnostic output.
+                // println(s"$configDescription\tsObs=$sObs\tsMean=$sMean")
+                if (sObs > sMean) {
+                  countStdObsGtStdMean += 1
+                } else {
+                  countStdObsLtStdMean += 1
+                }
+              }
+
+              // We have strong theoretical guarantees on the behavior of GuessTheMeanLearner, so let's exercise them.
+              // NOTE: these bounds reflect a ~3x systematic variance under-estimation in this particular test setting.
+              {
+                val rtolLower = baseLearner match {
+                  case _: GuessTheMeanLearner => 3.5
```

Comment:
Did you intend to include `GuessTheMeanLearner` in the sequence of base learners?

Reply:
I would like to, but there's no such MultiTaskLearner. I cleaned this up to be a little more explicit about why the code looks like this when there's just one baseLearner.